### DISCUSSION TITLE
default behavior of tab

### DISCUSSION THREAD

**Discussion by blaisemGH** [PowerShell/PowerShell#17091] [Enhancements]
Hello,

QoL idea: What do you all think of setting the default tab button behavior in Windows systems to "Complete", e.g., equivalent to the command:

Set-PSReadlineKeyHandler -Key Tab -Function Complete

This is how it works in Bash. I don't know what the default is in PS Core on Unix systems, but on Windows the default function for tab is "TabCompleteNext".

I've never found a newcomer to PowerShell on a Windows OS who hasn't been annoyed by this obtuse default tab function. I've even see people immediately drop the shell and go back to the Windows Explorer GUI just because the shell navigation without a useful tab button is so essential. I have various dev colleagues used to Unix systems, and when I tell them to give PowerShell a go, this is one of the first hurdles that can scare them off.

Maybe a good corollary question: Does anyone use PowerShell to navigate their filesystem regularly _without_ tab being set to Complete (either as TabCompleteNext, Menu, or anything else if it exists)?

**Comment by SeeminglyScience** [comment]
> I don't know what the default is in PS Core on Unix systems

It's `Complete` on non-Windows platforms.

> Does anyone use PowerShell to navigate their filesystem regularly without tab being set to Complete

Yup! Probably depends largely on what you were first exposed to. If you were to watch me try to use `Complete`, it would probably look very similar to non-Windows folks trying to use `TabCompleteNext`. Hell some folks like it so much they even make their editor's intellisense behave the same way (and it's even the default for the ISE).

I see the exact opposite conversation occur when Windows folks try PowerShell on non-Windows for the first time. Thankfully it's a pretty easy customization.

**Comment by kasini3000** [comment]
are you looking for this ?

win：
pwsh -c 'Add-Content  -Value "Set-PSReadlineOption -EditMode Windows" -LiteralPath $profile '

linux bash:
pwsh -c 'Add-Content  -Value "Set-PSReadlineOption -EditMode  Emacs" -LiteralPath $profile '
